src/App.vue

```diff
@@ -163,54 +296,72 @@ import {
 import { useMissionStore } from '@/stores/mission'
 
 import AltitudeSlider from './components/AltitudeSlider.vue'
-import Dialog from './components/Dialog.vue'
 import EditMenu from './components/EditMenu.vue'
+import GlassButton from './components/GlassButton.vue'
 import MiniWidgetContainer from './components/MiniWidgetContainer.vue'
 import SlideToConfirm from './components/SlideToConfirm.vue'
 import Alerter from './components/widgets/Alerter.vue'
 import { datalogger } from './libs/sensors-logging'
+import { useAppInterfaceStore } from './stores/appInterface'
 import { useMainVehicleStore } from './stores/mainVehicle'
 import { useWidgetManagerStore } from './stores/widgetManager'
 
 const widgetStore = useWidgetManagerStore()
 const vehicleStore = useMainVehicleStore()
+const responsiveStore = useAppInterfaceStore()
 
 const showConfigurationMenu = ref(false)
 
 // Main menu
 const showMainMenu = ref(false)
+const isSlidingIn = ref(false)
+const isSlidingOut = ref(false)
+const mainMenuStep = ref(1)
 
-// When a isVehicleArmed change its value a watcher call Swal.close, this flag is
-// used to avoid closing others Swal instances instead of the one intended
 let requestDisarmConfirmationPopup = false
-const openMainMenu = (): void => {
-  if (!vehicleStore.isArmed) {
-    showMainMenu.value = true
-    return
-  }
 
-  requestDisarmConfirmationPopup = true
-  Swal.fire({
-    title: 'Be careful',
-    text: 'The vehicle is currently armed, it is not recommended to open the main menu.',
-    icon: 'warning',
-    showCancelButton: true,
-    showConfirmButton: true,
-    cancelButtonText: 'Continue anyway',
-    confirmButtonText: 'Disarm vehicle',
-  }).then((result) => {
-    // Opens the main menu only after disarming by the slider is confirmed
-    if (result.isConfirmed && vehicleStore.isArmed) {
-      vehicleStore.disarm().then(() => {
-        showMainMenu.value = true
-      })
-    } else if (result.dismiss === Swal.DismissReason.cancel) {
-      showMainMenu.value = true
-    }
-    requestDisarmConfirmationPopup = false
-  })
+const toggleMainMenu = useDebounceFn(() => {
```

Comment:
I think we are forgetting to add the arm check back here.

Reply:
I did remove it on purpose, actually.
Do you think it is still necessary to warn about this, now that the menu and the configuration screens are way smaller, not-centered and translucent?

If so, we could replace the modal warning by a snackbar and a system warning that doesn't need an user action to be closed. My concern on keeping the warning there is that it adds friction and increases the time needed for the user to make the intended change on settings, for example.

Reply:
I think we should discuss that with the team. Maybe open a thread in Slack? The warn is not just about the fact that the pilot cannot see the screen, but also because there are a lot of configurations that can make the user lose contact with the vehicle, lose control, etc, that can cause a harming situation.

If you prefer to postpone the discussion I think we can keep the warning an open a separate issue for that.

Reply:
Agreed, I'll return the warning until we decide abou that.